### DISCUSSION TITLE
Hide a field in the Infolist builder by using a closure

### DISCUSSION THREAD

**Discussion by Kovah** [filamentphp/filament#8764] [Help] [answered]
I am converting some old Filament 2 views to the new Infolist builder. However, it seems that using a closure in the `hidden()` field is not working (yet).

When using a closure in the hidden() method like this
```php
Infolists\Components\TextEntry::make('published_at')
    ->hidden(fn (Get $get) => $get('status') !== 'published'),
```
I am receiving the following error:
```
Typed property Filament\Forms\Components\Component::$container must not be accessed before initialization
```

Also, there seems to be no `Filament\Infolist\Get`, only `Filament\Forms\Get`.

Has anyone made closures work in the Infolist builder?

**Comment by leandrocfe** [comment, marked as answer]
```php
->hidden(fn ($record): bool => $record->status !== 'published')
```

**Reply by Kovah** [reply]
Oh thanks, this worked! Is this documented somewhere?

**Reply by leandrocfe** [reply]
> Oh thanks, this worked! Is this documented somewhere?

Not sure, but you can send a PR if you want.
